### PR TITLE
Adding terms for passwordless

### PR DESCRIPTION
**Pull request by luisrudge**
fix https://github.com/auth0/lock/issues/1421

### CODE REVIEW

**Comment by sholladay** [comment]
Any thoughts on when this might land? I'm considering switching to passwordless, but would need to require new users accept our Terms of Service during signup.

**Comment by luisrudge** [comment]
@sholladay Hi! I still need to add the tests for this. Hopefully I can get some time on this issue this week or the next.

**Comment by GituBojan** [comment]
@luisrudge Have you been able to look into that? Thank you!

File: src/connection/passwordless/actions.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ import {
   send,
   setPasswordlessStarted,
   setResendFailed,
-  setResendSuccess
+  setResendSuccess,
+  toggleTermsAcceptance as switchTermsAcceptance
```
Comment (joshcanhelp) [inline src/connection/passwordless/actions.js]: Would `switchTermsAcceptance` be better named as `toggleTermsAcceptanceParent` or similar? It looks like you're overriding that method here but the name suggests it does something different ... and even then, if it did, the name is pretty similar.

File: src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap [hunk 2]
```diff
@@ -59,3 +82,27 @@ Array [
...
+
+exports[`passwordless connection mustAcceptTerms() should return \`mustAcceptTerms\` 1`] = `
+Array [
+  "model",
+  "mustAcceptTerms",
+  false,
+]
+`;
+
+exports[`passwordless connection toggleTermsAcceptance() should tset \`termsAccepted\` to false when \`termsAccepted\` is true 1`] = `
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap]: Typo "tset"

File: src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap [hunk 3]
```diff
@@ -59,3 +82,27 @@ Array [
...
+Array [
+  "model",
+  "termsAccepted",
+  true,
+]
+`;
+
+exports[`passwordless connection toggleTermsAcceptance() should tset \`termsAccepted\` to true when \`termsAccepted\` is false 1`] = `
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap]: Typo "tset"

File: src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap [hunk 4]
```diff
@@ -1,10 +1,31 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`passwordless connection initPasswordless() calls initNS  with mustAcceptTerms:false when opts.mustAcceptTerms is false 1`] = `
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap]: Extra " "

File: src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap [hunk 5]
```diff
@@ -1,10 +1,31 @@
...
+Array [
+  null,
+  Immutable.Map {
+    "send": "code",
+    "mustAcceptTerms": false,
+  },
+]
+`;
+
+exports[`passwordless connection initPasswordless() calls initNS  with mustAcceptTerms:true when opts.mustAcceptTerms is true 1`] = `
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap]: Extra " "

File: src/__tests__/connection/passwordless/passwordless.test.js [hunk 6]
```diff
@@ -51,6 +51,18 @@ describe('passwordless connection', () => {
         });
         expectMockToMatch(mockFns.initNS, 1);
       });
+      it('with mustAcceptTerms:true when opts.mustAcceptTerms is true', () => {
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/passwordless.test.js]: Awkward statement ... should start with "should"

File: src/__tests__/connection/passwordless/passwordless.test.js [hunk 7]
```diff
@@ -51,6 +51,18 @@ describe('passwordless connection', () => {
...
+        initPasswordless(null, {
+          mustAcceptTerms: true
+        });
+        expectMockToMatch(mockFns.initNS, 1);
+      });
+      it('with mustAcceptTerms:false when opts.mustAcceptTerms is false', () => {
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/passwordless.test.js]: Awkward statement ... should start with "should"

File: src/__tests__/connection/passwordless/passwordless.test.js [hunk 8]
```diff
@@ -71,4 +83,23 @@ describe('passwordless connection', () => {
...
   });
+  describe('mustAcceptTerms()', () => {
+    it('should return `mustAcceptTerms`', () => {
+      require('connection/passwordless/index').mustAcceptTerms('model');
+      expectMockToMatch(mockFns.get, 1);
+    });
+  });
+
+  describe('toggleTermsAcceptance()', () => {
+    it('should tset `termsAccepted` to false when `termsAccepted` is true', () => {
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/passwordless.test.js]: Type "tset"

File: src/__tests__/connection/passwordless/passwordless.test.js [hunk 9]
```diff
@@ -71,4 +83,23 @@ describe('passwordless connection', () => {
...
+      mockFns.get.mockReturnValue(true);
+      require('connection/passwordless/index').toggleTermsAcceptance('model');
+      expectMockToMatch(mockFns.tset, 1);
+    });
+    it('should tset `termsAccepted` to true when `termsAccepted` is false', () => {
```
Comment (joshcanhelp) [inline src/__tests__/connection/passwordless/passwordless.test.js]: Type "tset"

File: src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap (hunk 5 above)
Comment (luisrudge) [reply src/__tests__/connection/passwordless/__snapshots__/passwordless.test.js.snap]: that's just the test runner joining a describe+it. not much I can do

File: src/connection/passwordless/actions.js (hunk 1 above)
Comment (luisrudge) [reply src/connection/passwordless/actions.js]: renamed to `internalToggleTermsAcceptance`

File: src/__tests__/connection/passwordless/passwordless.test.js (hunk 6 above)
Comment (luisrudge) [reply src/__tests__/connection/passwordless/passwordless.test.js]: it makes sense if you join the `describe` description with this (that's how the test runner shows it).

`calls initNS with mustAcceptTerms:true when opts.mustAcceptTerms is true`

**Pull request merged**